test-plans/Signon.scala

```diff
@@ -0,0 +1,40 @@
+package govuk
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+
+/**
+ * Signon scenario singleton object.
+ *
+ * Env vars USERNAME and PASSWORD are used to authenticate with the signon
+ * application configured using the JAVA_OPT 'signonUrl'.
+ * Call 'exec(Signon.authenticate)' to use in a Gatling scenario.
+ */
+object Signon {
+  val signonBaseUrl = sys.props.get("signonUrl").get
+  val signonUrl = s"$signonBaseUrl/users/sign_in"
+
+  val authenticate = exec(
+    http("Visit signon")
+      .get(signonUrl)
+      .check(status.is(200))
+      .check(
+        regex("Sign in to GOV.UK").exists,
+        css("input[name=authenticity_token]", "value").saveAs("signonAuthToken")
+      )
+  )
+  .exec(
+    http("Authenticate in signon")
+      .post(signonUrl)
+      .formParam("authenticity_token", """${signonAuthToken}""")
+      .formParam("user[email]", sys.env.get("USERNAME").get)
```

Comment:
Is there a reason these are environment variables while everything else is a property?

Reply:
I don't suppose there is really, possibly due to how I was running the scenario: setting the creds in a distinct step, then exporting the properties as I ran the scenario. There's no reason why they can't be properties though. I'll update for consistency.

Reply:
@barrucadu ah I remember now, yes there was a reason, spaces in JAVA_OPTS properties are problematic, at least for Gatling.
It looks like the Gatling startup script doesn't handle these correctly.
For example the property `-Dpassword="my information is secret"` results in `java.lang.ClassNotFoundException: information`.

Reply:
Ah, shell scripts...